Content.Server/Chemistry/ReactionEffects/SmokeReactionEffect.cs

```diff
@@ -0,0 +1,61 @@
+﻿using System;
+using Content.Server.Explosions;
+using Content.Server.GameObjects.Components.Chemistry;
+using Content.Server.Interfaces.Chemistry;
+using Content.Server.Utility;
+using Content.Shared.Audio;
+using Robust.Server.GameObjects.EntitySystems;
+using Robust.Shared.GameObjects.Systems;
+using Robust.Shared.Interfaces.GameObjects;
+using Robust.Shared.Interfaces.Map;
+using Robust.Shared.IoC;
+using Robust.Shared.Map;
+using Robust.Shared.Serialization;
+
+namespace Content.Server.Chemistry.ReactionEffects
+{
+    public class SmokeReactionEffect : IReactionEffect
+    {
+        /// <summary>
+        /// Used for calculating the spread range of the smoke based on the intensity of the reaction.
+        /// </summary>
+        private float _rangeMultiplier;
+        /// <summary>
+        /// How many seconds will the smoke stay, counting after fully spreading.
+        /// </summary>
+        private float _duration;
+        /// <summary>
+        /// How many seconds between each spread step.
+        /// </summary>
+        private float _spreadDelay;
+
+        private string _smokeSound;
+
+        public void ExposeData(ObjectSerializer serializer)
+        {
+            serializer.DataField(ref _rangeMultiplier, "rangeMultiplier",0.2f);
+            serializer.DataField(ref _duration, "duration", 10f);
+            serializer.DataField(ref _spreadDelay, "spreadDelay", 0.5f);
+            serializer.DataField(ref _smokeSound, "smokeSound",string.Empty);
+        }
+
+        public void React(IEntity solutionEntity, double intensity)
+        {
+            var contents = solutionEntity.GetComponent<SolutionContainerComponent>();
```

Comment:
Instead of using GetComponent, which throws an exception if that component is not there, use TryGetComponent, report an error with Logger.Error then return out, so that it doesn't crash the server if something is messed up. Same for GetComponent< SmokeComponent >().